omod/src/main/java/org/openmrs/module/reporting/web/controller/portlet/CohortIndicatorAndDimensionSpecificationPortletController.java

```diff
@@ -60,17 +61,23 @@ protected void populateModel(HttpServletRequest request, Map<String, Object> mod
        	if (StringUtils.isEmpty(mappedUuid)) { // If you are not selecting a new Indicator, get the currently saved specification for the passed index
        		if (specification.getIndicator() != null) {
        			mappedObj = specification.getIndicator().getParameterizable();
-    			mappings = specification.getIndicator().getParameterMappings();
+    			mappings = specification.getIndicator().getParameterMappings();   			
     		}
        	}
        	else if (mappedUuid != null) { // If you are selecting a new Indicator, get it
-       		mappedObj = (CohortIndicator)DefinitionContext.getIndicatorService().getDefinitionByUuid(mappedUuid);
-       	}
-       	model.put("mappedObj", mappedObj);
-       	model.put("mappings", mappings);
+       		Object testObj = DefinitionContext.getIndicatorService().getDefinitionByUuid(mappedUuid);
+       		if (testObj instanceof CohortIndicator) {
+       			mappedObj = (CohortIndicator)DefinitionContext.getIndicatorService().getDefinitionByUuid(mappedUuid);
+       		} 
+       	} 	
+       	
+       	if (mappedObj != null) {
+       		model.put("mappedObj", mappedObj);

```

Comment:
No reason to check for null here, just put it in the model regardless like was done previously